plugins/pagerduty-incidents/src/components/PagerDutyIncidents.tsx

```diff
@@ -0,0 +1,384 @@
+import React, { useState, useEffect, useMemo } from "react";
+import { PluginContextLocation } from "@cortexapps/plugin-core";
+import {
+  SimpleTable,
+  // Text,
+  Title,
+  Loader,
+  usePluginContext,
+} from "@cortexapps/plugin-core/components";
+
+import {
+  Avatar,
+  AvatarGroup,
+  Box,
+  Badge,
+  Flex,
+  // Center,
+  Text,
+} from "@chakra-ui/react";
+
+import "../baseStyles.css";
+
+interface PagerDutyIncidentsProps {
+  entityYaml: Record<string, any>;
+}
+
+const StatusBadge: React.FC<{ status: string }> = ({ status }) => {
+  const color =
+    status === "resolved"
+      ? "green"
+      : status === "acknowledged"
+      ? "yellow"
+      : "red";
+  return (
+    <Badge p={1} colorScheme={color} variant="subtle">
+      {status}
+    </Badge>
+  );
+};
+
+const UrgencyBadge: React.FC<{ urgency: string }> = ({ urgency }) => {
+  const variant = urgency === "high" ? "solid" : "subtle";
+  return (
+    <Badge p={1} variant={variant}>
+      {urgency}
+    </Badge>
+  );
+};
+
+const PagerDutyIncidents: React.FC<PagerDutyIncidentsProps> = ({
+  entityYaml,
+}) => {
+  const [hasIncidents, setHasIncidents] = useState(false);
+  const context = usePluginContext();
+
+  const [service, setService] = useState(null as any);
+  const [oncalls, setOncalls] = useState([]);
+
+  const [incidents, setIncidents] = React.useState([]);
+  const [isLoading, setIsLoading] = React.useState(
+    context.location === PluginContextLocation.Entity
+  );
+
+  const [errorMessage, setErrorMessage] = React.useState("");
+
+  const pdId: string = useMemo(
+    () => entityYaml?.info?.["x-cortex-oncall"]?.pagerduty?.id || "",
+    [entityYaml?.info]
+  );
+  const pdType: string = useMemo(
+    () => entityYaml?.info?.["x-cortex-oncall"]?.pagerduty?.type || "",
+    [entityYaml?.info]
+  );
+
+  useEffect(() => {
+    const fetchService = async (): Promise<void> => {
```

Comment:
Would it be better to separate these fetches into their own hooks so that if something fails they'll be isolated? Thinking splitting the fetch logic out into its own components, so fetching on call, services, and incidents.